### PR TITLE
Feature create user account 164674064

### PR DESCRIPTION
**Pull request by neelxie**
[#164674064]Create user account
#### What does this PR do
 - Sign up feature for the app.
 - Get valid user and add them to app else reject.
#### Description of task to be completed.
 - A user is supposed to sign up with their credentials to get an account
 - Wrong details and the user will get relevant error message. 
 - A user recieves a token as confirmation
#### Screenshots
pr-164674064

#### You can check out the relevant pivotal tracker story [here](https://www.pivotaltracker.com/n/projects/2318759/stories/164674064)
@bekeplar @jennizalwango @alexxsanya @Wabuluka @kyakusahmed @ivankivu @Phionanamugga @nanfuka @HabibSentongo

### CODE REVIEW

File: app/controller/user_controller.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+from flask import request
+import datetime
+import jwt
+from ..utils.validation import Valid
+from ..utils.auth import my_secret_key
+
+from ..models.user_model import (
+    Base, User, UserDB)
+
+class User_Controller:
```
Comment (byarustev) [inline app/controller/user_controller.py]: remove the underscore.

File: app/models/user_model.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+""" This is the Users' models file."""
+from ..utils.validation import Valid
+
+class Base:
```
Comment (byarustev) [inline app/models/user_model.py]: This name can be made Person. Base is not descriptive, if it wasn't for the comment I would not have known what It was about.

File: app/views/app_views.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# from .mail_views import mail_bp
+from .user_views import auth_bp
+
+def create_app():
+    """
+    App factory for the epic mail app.
+    """
+    app = Flask(__name__)
+    app.register_blueprint(auth_bp, url_prefix='/api/v1/auth')
+    # app.register_blueprint(mail_bp, url_prefix='/api/v1')
```
Comment (byarustev) [inline app/views/app_views.py]: what is this for? please remove commented functionality. If a comment is not describing what is taking place should not be committed to git

**Review by byarustev — CHANGES_REQUESTED** [review]

**Comment by neelxie** [comment]
Thank you. Let me work on that and let you know.

**Pull request merged**